### PR TITLE
[new build] use rollup, add es + umd builds. fixes #143

### LINKED ISSUE #143: Add module support for tree shaking
Add support for the `"module"` field in package.json so that webpack/rollup can do proper tree shaking. If you use @vx currently with Rollup, you will get an error by default due to the only availability being commonjs that it cannot understand, so you have to add all the exports of each sub-library of @vx explicitly in rollup.config.js:

https://github.com/rollup/rollup-plugin-commonjs#custom-named-exports

The result is that if you import one part of a `@vx/shape`, for example, you have to import its entirety.

I believe this can be as simple as building each library twice: Once as it currently is (default es5 transpiling by babel), and a second build into another folder, such as `module`, where the `modules` flag is turned off for babel in the babelrc:
```
"presets": [["es2015", { "modules": false }], "react", "stage-0"],
```
https://babeljs.io/docs/plugins/preset-es2015/#options

### PR DESCRIPTION
**Pull request by hshoff**
fixes: https://github.com/hshoff/vx/issues/143

- [x] adds `dist/vx-{name}.{umd,es}.js` files
- [x] adds `module` field to `package.json` that points to `dist/vx-{name}.es.js` for bundlers that support it (webpack, rollup, etc)
- [x] `main` field points to `dist/vx-{name}.umd.js`
- [x] removes [`react-fatigue-dev`](https://github.com/tj/react-fatigue-dev) + Makefile build (means you can build on windows now)
- [x] build targets  `"android": 30, "chrome": 35, "edge": 14, "explorer": 9, "firefox": 52, "safari": 8, "ucandroid": 1` (matching [airbnb-babel-preset](https://github.com/airbnb/babel-preset-airbnb/blob/master/index.js#L9-L17)
- [x] keeps `build/` with `cjs` babel build files for [deep paths](https://github.com/hshoff/vx/issues/143#issuecomment-367649444) (no breaking changes 🤞)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,8 +8,7 @@
     "docs": "node ./scripts/docs/index.js",
     "prepare-release": "git checkout master && git pull --rebase origin master && npm run docs && lerna updated",
     "release": "npm run prepare-release && lerna publish --exact",
-    "format": "prettier \"{packages,scripts}/**/*.js\" --write",
-    "precommit": "lint-staged"
```
Comment (hshoff) [inline package.json]: TODO: add this back, it was blowing up on my Win10 box

**Comment by williaster** [comment]
this is awesome! 🙌

File: package.json (hunk 1 above)
Comment (sto3psl) [reply package.json]: This looks amazing!

Don't know if it applies but you can ignore the commit hooks if you do `git commit -m "message" -n`.

Comment (hshoff) [reply package.json]: @sto3psl awesome! didn't know I could do that... for context it was only blowing up when i did a commit on windows

**Comment by hshoff** [comment]
going to merge this and release it as `0.0.165-beta.*`

**Pull request merged**